config/opal_check_libnl.m4

```diff
@@ -0,0 +1,223 @@
+dnl -*- shell-script -*-
+dnl
+dnl Copyright (c) 2015      Research Organization for Information Science
+dnl                         and Technology (RIST). All rights reserved.
+dnl $COPYRIGHT$
+dnl
+dnl Additional copyrights may follow
+dnl
+dnl $HEADER$
+dnl
+
+dnl
+dnl Helper function to parse --with-libnl* options
+dnl
+dnl $1: variable name
+dnl $2: --with-<foo> value
+dnl
+AC_DEFUN([OPAL_LIBNL_PARSE_WITH],[
+	case "$2" in
+	no)
+		# Nope, don't want it
+		opal_want_$1=no
+		;;
+	yes)
+		# Yes, definitely want it
+		opal_want_$1=yes
+		;;
+	default)
+		# Default case -- try and see if we can find it
+		opal_want_$1=default
+		opal_$1_location=/usr
+		;;
+	*)
+		# Yes, definitely want it -- at a specific location
+		opal_want_$1=yes
+		opal_$1_location="$2"
+		;;
+	esac
+])
+
+dnl OPAL_LIBNL_SANITY_INIT()
+dnl --------------------------------------------------------------------
+AC_DEFUN([OPAL_LIBNL_SANITY_INIT], [
+    opal_libnl_version=0
+    opal_libnl_libs=
+    opal_libnl3_libs=
+    AC_ARG_WITH([libnl],
+                [AC_HELP_STRING([--with-libnl(=DIR)],
+                                [Directory prefix for libnl (typically only necessary if libnl is installed in a location that the compiler/linker will not search by default)])],
+                [], [with_libnl=default])
+
+    # The --with options carry two pieces of information: 1) do
+    # you want a specific version of libnl, and 2) where that
+    # version of libnl lives.  For simplicity, let's separate
+    # those two pieces of information.
+    OPAL_LIBNL_PARSE_WITH([libnl], [$with_libnl])
+])
+
+dnl OPAL_LIBNL_SANITY_CHECK(lib, function, LIBS)
+dnl --------------------------------------------------------------------
+AC_DEFUN([OPAL_LIBNL_SANITY_CHECK], [
+    case $host in
+        *linux*)
+            AC_LANG_PUSH(C)
+            cat > conftest_c.$ac_ext << EOF
+extern void $2 (void);
+int main(int argc, char *argv[]) {
+    $2 ();
+    return 0;
+}
+EOF
+            OPAL_LOG_COMMAND([$CC -o conftest $CFLAGS $CPPFLAGS conftest_c.$ac_ext $LDFLAGS -l$1 $LIBS $3],
+                             [ldd_output=`ldd conftest`
+                              libnl_version=0
+                              AS_IF([echo $ldd_output | grep -q libnl.so],
+                                    [AS_IF([test $opal_libnl_version -eq 3],
+                                           [AC_MSG_WARN([lib nl version conflict: $opal_libnl3_libs requires libnl-3 whereas $1 requires libnl])],
+                                           [opal_libnl_libs="$opal_libnl_libs $1"
+                                            opal_libnl_version=1])
+                                     libnl_version=1])
+                              AS_IF([echo $ldd_output | grep -q libnl-3.so],
+                                    [AS_IF([test $libnl_version -eq 1],
+                                           [AC_MSG_ERROR([lib $1 requires both libnl and libnl3\nthis is just insane\nAborting...])])
```

Comment:
This doesn't output like you would expect it to.  Here's what I get:

```
configure: error: lib fabric requires both libnl and libnl3\nthis is just insane\nAborting...
```

I think you probably want something like this:

```
AC_MSG_WARN([$1 requires both libnl and libnl3 -- yoinks!])
AC_MSG_WARN([This is a configuration that is known to cause run-time crashes])
AC_MSG_ERROR([Cannot continue])
```